kratos/integration/cad_integration_domain.cpp

```diff
@@ -0,0 +1,370 @@
+            shape_function_derivatives_order = rParameters["shape_function_derivatives_order"].GetInt();
+        }
+        else {
+            KRATOS_INFO_IF("CreateQuadraturePointGeometries", EchoLevel > 4)
+                << "shape_function_derivatives_order is not provided and thus being considered as 1. " << std::endl;
+        }
+
+        KRATOS_INFO_IF("CreateQuadraturePointGeometries", EchoLevel > 0)
+            << "Creating " << name << "s of type: " << type
+            << " for " << rQuadraturePointGeometryList.size() << " geometries"
+            << " in " << rCadSubModelPart.Name() << "-SubModelPart." << std::endl;
+
+        for (SizeType i = 0; i < rQuadraturePointGeometryList.size(); ++i)
+        {
+            GeometriesArrayType geometries;
+            rQuadraturePointGeometryList[i].CreateQuadraturePointGeometries(
+                geometries, shape_function_derivatives_order);
+
+            KRATOS_INFO_IF("CreateQuadraturePointGeometries", EchoLevel > 1)
+                << geometries.size() << " quadrature point geometries have been created." << std::endl;
+
+            if (type == "element" || type == "Element") {
+                int id = 0;
+                if (rCadSubModelPart.GetRootModelPart().Elements().size() > 0)
+                    id = rCadSubModelPart.GetRootModelPart().Elements().back().Id() + 1;
+
+                CadIntegrationDomain::CreateElements(geometries, rCadSubModelPart, name, id, EchoLevel);
+            }
+            else if (type == "condition" || type == "Condition") {
+                int id = 0;
+                if (rCadSubModelPart.GetRootModelPart().Conditions().size() > 0)
+                    id = rCadSubModelPart.GetRootModelPart().Conditions().back().Id() + 1;
+
+                CadIntegrationDomain::CreateConditions(geometries, rCadSubModelPart, name, id, EchoLevel);
+            }
+            else {
+                KRATOS_ERROR << "\"type\" does not exist: " << type
+                    << ". Possible types are \"element\" and \"condition\"." << std::endl;
+            }
+        }
+    }
+
+    void CadIntegrationDomain::CreateElements(
+        GeometriesArrayType& rQuadraturePointGeometryList,
+        ModelPart& rCadSubModelPart,
+        std::string& rElementName,
+        int& rIdCounter,
+        int EchoLevel)
+    {
+        const Element& rReferenceElement = KratosComponents<Element>::Get(rElementName);
+
+        ElementsContainerType new_element_list;
+
+        KRATOS_INFO_IF("CreateElements", EchoLevel > 2)
+            << "Creating " << rQuadraturePointGeometryList.size()
+            << " elements of type " << rElementName
+            << " in " << rCadSubModelPart.Name() << "-SubModelPart." << std::endl;
+
+        for (IndexType i = 0; i < rQuadraturePointGeometryList.size(); ++i)
+        {
+            auto p_element = rReferenceElement.Create(rIdCounter, rQuadraturePointGeometryList(i), nullptr);
```

Comment:
@pooyan-dadvand elements/ conditions can only be created with the Geometry::Pointer the other constructor does not work. (e.g.: rQuadraturePointGeometryList[i] does not work.)

We should figure out a way to prevent people from doing that. Or maybe fixing the problem, but I am not sure if that is possible.